src/components/TextArea/TextArea.tsx

```diff
@@ -28,40 +34,70 @@ export type TextAreaProps = {
   onKeyDown?: React.KeyboardEventHandler<HTMLTextAreaElement | HTMLInputElement>;
   /** Callback fired when the `input` value typed is changed */
   onInput?: React.EventHandler<any>;
-} & Pick<TextInputBaseProps, 'status'> &
+} & Pick<HTMLTextAreaElement, 'cols' | 'rows'> &
+  Pick<TextInputBaseProps, 'status' | 'label'> &
   TestProps;
 
 const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>((props, ref) => {
   const {
     id = undefined,
-    placeholder = '',
+    label,
+    placeholder,
     isRequired = false,
+    isReadOnly,
     isDisabled,
+    status,
     isResizeEnabled = true,
+    maxCharacters,
     ...rest
   } = props;
 
+  const isLocked = status?.type === 'read-only';
+
   const theme = useTheme();
-  const sx = sxProp(!isDisabled && isResizeEnabled, theme);
+  const sx = sxProp({
+    isResizeEnabled: !isDisabled && !isLocked && isResizeEnabled,
+    label,
+    placeholder,
+  })(theme);
+
+  const shouldShowCounter = maxCharacters && status?.type != 'error' && !status?.hintMessage;
+
+  const counter = shouldShowCounter ? (
+    <div css={hintMessageStyle(theme)}>
+      {rest.value?.length}/{maxCharacters}
+    </div>
+  ) : undefined;
 
   return (
     <React.Fragment>
       <TextInputBase {...props} sx={sx}>
-        <div css={{ width: '100% ' }}>
+        <div css={{ width: '100%' }}>
           <textarea
+            readOnly={isLocked || isReadOnly}
```

Comment:
use `role="textbox"` and i think we should use `aria-multiline` as true because we don't have single like only